services/libs/tinybird/README.md

```diff
@@ -149,3 +149,81 @@ Until we move fully to **Tinybird Forward** (which will support migration script
 2. **Pipe-by-pipe updates** for zero downtime where #1 is not enough
 
 Pick the method that best fits your workflow and datasource complexity.
+
+# Testing Tinybird Pipes Locally
+
+This guide explains how to test a Tinybird data pipeline ("pipe") on your local Tinybird environment. We will fetch sample data (fixtures) from a staging Tinybird workspace and use it to run and verify a pipe locally. The steps below are written for a developer who may not be familiar with Tinybird, and they are organized in a clear, numbered format for easy follow-up.
+
+## Prerequisites
+
+- **Tinybird CLI and Local Environment:** Make sure you have the Tinybird CLI (`tb`) installed.  
+  To start a local Tinybird instance, run the following Docker command:
+  ```bash
+  docker run --platform linux/amd64 -p 7181:7181 --name tinybird-classic-local     -e COMPATIBILITY_MODE=1 -d tinybirdco/tinybird-local:latest
+  ```
+  This will start Tinybird Local at `http://localhost:7181`.
+
+- **Staging Workspace Access:** You need access to a Tinybird staging (or production) workspace with an API token that has permission to read data (we'll use it to export data).  
+- **Project Files:** Ensure you have your Tinybird project files available locally – this includes the data source definitions and pipe (.pipe) files you plan to test. For example, if you plan to test a pipe that uses a data source named `insightsProjects`, you should have the corresponding `insightsProjects.datasource` file (or have created that data source) in your local workspace. Similarly, have the pipe file (e.g. `activities_filtered.pipe` and any related pipes) ready in your project directory.
+
+## Steps to Test the Tinybird Integration Locally
+
+1. **Set Up Environment Variables and Authenticate**  
+   First, set up environment variables for both your local Tinybird instance and the staging environment, then authenticate using the Tinybird CLI. This will ensure the CLI is pointed to the correct environment when running commands. Replace `<YOUR_STAGING_TOKEN>` with your actual staging API token. The local admin token is retrieved via the local Tinybird API:
+   ```bash
+   # Set Tinybird local API endpoint and get the local admin token
+   export TB_HOST_LOCAL="http://localhost:7181"
+   TB_TOKEN_LOCAL="$(curl -s "$TB_HOST_LOCAL/tokens" | jq -r '.workspace_admin_token')"
+
+   # Set Tinybird staging API endpoint and token (replace with your staging info)
+   export TB_HOST_STAGING="https://api.us-west-2.aws.tinybird.co"   # use the correct region/domain for your case
+   export TB_TOKEN_STAGING="<YOUR_STAGING_TOKEN>"
```

Comment:
I would suggest not telling people to run commands containing secret tokens in their shell because the token will most likely be recorded in the shell history.

Instead, perhaps we could tell people to have a `.env` file that they could `source` before running the `tb` command. The effect would be the same (env vars available in the shell) but the token would not be recorded in the shell history file.

Alternatively, though it seems too much for this, we could also suggest the use of [dotenvx](https://dotenvx.com/).